tests/functional/builtins/codegen/test_abi_decode_fuzz.py

```diff
@@ -0,0 +1,371 @@
+        return tuple(_go(item_t) for item_t in typ.member_types)
+
+    if isinstance(typ, SArrayT):
+        return [_go(typ.value_type) for _ in range(typ.length)]
+
+    if isinstance(typ, DArrayT):
+        n = draw(st.integers(min_value=0, max_value=typ.length))
+        return [_go(typ.value_type) for _ in range(n)]
+
+    if isinstance(typ, StringT):
+        # technically the ABI spec doesn't say string has to be valid utf-8,
+        # but eth-stdlib won't encode invalid utf-8
+        return draw(st.text(max_size=typ.length))
+
+    if isinstance(typ, BytesT):
+        return draw(st.binary(max_size=typ.length))
+
+    if isinstance(typ, IntegerT):
+        lo, hi = typ.ast_bounds
+        return draw(st.integers(min_value=lo, max_value=hi))
+
+    if isinstance(typ, BytesM_T):
+        return draw(st.binary(min_size=typ.length, max_size=typ.length))
+
+    if isinstance(typ, BoolT):
+        return draw(st.booleans())
+
+    if isinstance(typ, AddressT):
+        ret = draw(st.binary(min_size=20, max_size=20))
+        return "0x" + ret.hex()
+
+    raise RuntimeError("unreachable")
+
+
+def _sort2(x, y):
+    if x > y:
+        return y, x
+    return x, y
+
+
+@st.composite
+def _mutate(draw, payload, max_mutations=MAX_MUTATIONS):
+    # do point+bulk mutations,
+    # add/edit/delete/splice/flip up to max_mutations.
+    if len(payload) == 0:
+        return
+
+    ret = bytearray(payload)
+
+    # for add/edit, the new byte is any character, but we bias it towards
+    # bytes already in the payload.
+    st_any_byte = st.integers(min_value=0, max_value=255)
+    payload_nonzeroes = list(x for x in payload if x != 0)
+    if len(payload_nonzeroes) > 0:
+        st_existing_byte = st.sampled_from(payload)
+        st_byte = st.one_of(st_existing_byte, st_any_byte)
+    else:
+        st_byte = st_any_byte
+
+    # add, edit, delete, word, splice, flip
+    possible_actions = "adwww"
+    actions = draw(st.lists(st.sampled_from(possible_actions), max_size=MAX_MUTATIONS))
+
+    for action in actions:
+        if len(ret) == 0:
+            # bail out. could we maybe be smarter, like only add here?
+            break
+
+        # for the mutation position, we can use any index in the payload,
+        # but we bias it towards indices of nonzero bytes.
+        st_any_ix = st.integers(min_value=0, max_value=len(ret) - 1)
+        nonzero_indexes = [i for i, s in enumerate(ret) if s != 0]
+        if len(nonzero_indexes) > 0:
+            st_nonzero_ix = st.sampled_from(nonzero_indexes)
+            st_ix = st.one_of(st_any_ix, st_nonzero_ix)
+        else:
+            st_ix = st_any_ix
+
+        ix = draw(st_ix)
+
+        if action == "a":
+            ret.insert(ix, draw(st_byte))
+        elif action == "e":
+            ret[ix] = draw(st_byte)
+        elif action == "d":
+            ret.pop(ix)
+        elif action == "w":
+            # splice word
+            st_uint256 = st.integers(min_value=0, max_value=2**256 - 1)
```

Comment:
i'm not sure how useful this range is. the interesting values will be on it's boundaries but we can sample those for "cheaper"

in most cases we'll just sample some gigantic random number and i'd argue it will just hit the same code path each time (and for ptrs will mostly likely just oog)